build-examples.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -e
+
+EXAMPLES=`find examples/ -maxdepth 1 -type d | tail -n +2`
+
+for example in $EXAMPLES; do
```

Comment:
I'd copy all of the examples to a temporary location; the clean is not real real real.

also instead of `$EXAMPLES` you can do `for example in examples/*`